cover-icon-element.js

```diff
@@ -2,31 +2,34 @@ class CoverIconElement extends HTMLElement {
 
   defaults = {
     breakpoints: [1, 50, 75, 100],
-    color: "#4d4d4d"
+    color: "#4d4d4d",
+    opacity: '0.3'
   }
 
   setImage(hass) {
     if (hass && this._config) {
-      let image = CoverIconElement.dataImage4(this._config.color);
+      let image = CoverIconElement.dataImage4(this._config.color, this._config.opacity);
 
       if (this._config.entity) {
         const stateObj = hass.states[this._config.entity];
         if (stateObj) {
           const position = stateObj.attributes.current_position;
           const state = stateObj.state;
 
-          if (state === "closing") {
+          if (state === "unavailable") {
```

Comment:
What do you think of re-using the same error icon for unknown entities as well?
```suggestion
          if (state === "unavailable" || state === "unknown") {
```

Reply:
Right, I planned to check for all the possible states of a cover element and add them later.

Reply:
Made a wrong assumption here since I tested this in the developer tools template. We won't ever get a state=unknown here. Maybe something like this?

```
let image = CoverIconElement.dataImageError(this._config.color);
if (this._config.entity) {
  const stateObj = hass.states[this._config.entity];
  if (stateObj) {
    const position = stateObj.attributes.current_position;
    const state = stateObj.state;
    if (state === "unavailable") {
      image = CoverIconElement.dataImageError(this._config.color);
    } else if (state === "closing") {
      image = CoverIconElement.dataImageClosing(this._config.color);
    } else if(state === "opening") {
      image = CoverIconElement.dataImageOpening(this._config.color);
    } else if (position >= this._config.breakpoints[3]) {
      image = CoverIconElement.dataImage0(this._config.color, this._config.opacity);
    } else if (position >= this._config.breakpoints[2]) {
      image = CoverIconElement.dataImage1(this._config.color, this._config.opacity);
    } else if (position >= this._config.breakpoints[1]) {
      image = CoverIconElement.dataImage2(this._config.color, this._config.opacity);
    } else if (position >= this._config.breakpoints[0]) {
      image = CoverIconElement.dataImage3(this._config.color, this._config.opacity);
    } else {
      image = CoverIconElement.dataImage4(this._config.color, this._config.opacity);
    }
  }
}
```

Reply:
That's what I suspected but didn't have time to test.
What if it's set to the alert unless in one of the "known" states? But maybe we are still missing some like maybe "tilting" or something.
Definitely need to check the docs for the cover element.

Reply:
>What if it's set to the alert unless in one of the "known" states?

Yep, that's what the above snippet does.

>But maybe we are still missing some like maybe "tilting" or something.

Yes, we can do that in the future but it shouldn't affect the current setup of showing "custom states" (opening/closing/unavailable) if there's an Icon mapping, and falling back to the position otherwise.